jib-core/src/test/java/com/google/cloud/tools/jib/http/WithServerFailoverHttpClientTest.java

```diff
@@ -151,6 +151,21 @@ public void testProxyCredentialProperties()
     }
   }
 
+  @Test
+  public void testClosingResourcesMultipleTimes_noErrors()
+      throws IOException, InterruptedException, GeneralSecurityException, URISyntaxException {
+    FailoverHttpClient httpClient = new FailoverHttpClient(true /*insecure*/, false, logger);
+    try (TestWebServer server = new TestWebServer(false, 2);
+        Response ignored1 = httpClient.get(new URL(server.getEndpoint()), request);
+        Response ignored2 = httpClient.get(new URL(server.getEndpoint()), request)) {
+      ignored1.close();
+      ignored2.close();
```

Comment:
Do we need to also write a test for transportsCreated?

Reply:
Actual transports are automatically created behind the scene (two transports in this test), and `httpClient.shutDown()` here shows there is no problem calling `HttpTransport.shutdown()` multiple times. The other test in `FailoverHttpClient` actually verifies if `httpClient.showDown()` calls `HttpTransport.shutdown()`, so I think I've covered everything?

Reply:
I'm going to merge this, but if you want to see something specific fixed, feel free to update, and I'll fix it.